craft_parts/plugins/base.py

```diff
@@ -122,3 +127,176 @@ def _get_java_post_build_commands(self) -> list[str]:
         # pylint: enable=line-too-long
 
         return link_java + link_jars
+
+
+class BasePythonPlugin(Plugin):
+    """A base class for Python plugins.
+
+    Provides common methods for dealing with Python items.
+    """
+
+    @override
+    def get_build_snaps(self) -> set[str]:
+        """Return a set of required snaps to install in the build environment."""
+        return set()
+
+    @override
+    def get_build_packages(self) -> set[str]:
+        """Return a set of required packages to install in the build environment.
+
+        Child classes that need to override this should extend the returned set.
+        """
+        return {"findutils", "python3-dev", "python3-venv"}
+
+    @override
+    def get_build_environment(self) -> dict[str, str]:
+        """Return a dictionary with the environment to use in the build step.
+
+        Child classes that need to override this should extend the dictionary returned
+        by this class.
+        """
+        python3_path = shutil.which("python3")
+        if python3_path is None:
+            raise errors.PluginEnvironmentValidationError(
```

Comment:
is this check new? isn't it going to conflict with `PARTS_PYTHON_INTERPRETER`?

Reply:
Hmm good point. This was pulled out of my initial poetry plugin. I'd like to check for an installed Python at some point, but that's out of scope here so I'm just going to remove this for now.